extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala

```diff
@@ -131,4 +144,39 @@ object SparkRangerAdminPlugin extends RangerBasePlugin("spark", "sparkSql")
       }
     }
   }
+
+  def getRangerPlugin(catalog: Option[String] = None): RangerBasePlugin = {
```

Comment:
def getOrCreate()

Reply:
shall we synchronize it?

Reply:
ok, let's synchronize it.
Plus,  `catalog2pluginMap` itself is already using `TrieMap` as implementation , which is a concurrent threadsafe lock-free concurrent.